### ISSUE TITLE
Fingerprint authentication no longer works [Mega 10.0.1]

### ISSUE THREAD

**Issue by repsorp** [meganz/android#228]
Mega 10.0.1 doesn't require fingerprint authentication or passcode on app startup despite option being activated

Mega 10.0.1
Android 12

![Screenshot_20231103-191911.png](https://github.com/meganz/android/assets/6297032/c8695382-0b0a-4897-8a33-eec2a75890fe)

**Comment by Perkolator** [comment]
Same here. It has also happened with older versions. The culprit seems to be updating the app. After user has set passcode (and additionally fingerprint?) and the app updates, the passcode/fingerprint is not asked anymore, even if in the settings the passcode is enabled.

Very very bad bug. Need to be fixed ASAP.

**Comment by repsorp** [comment]
10.0.2 update
fingerprint authentication still broken...

**Comment by Perkolator** [comment]
Same here. Both, just passcode (I'm using alphanumerical) or additional fingerprint lock doesn't work. Tried to disable whole passcode lock and re-enable it, no change.

**Comment by Perkolator** [comment]
Updated to version 10.0.3 and still doesn't work! Not passcode, nor fingerprint lock.

Steps I did to test this after update:

1. Open Mega app.
2. Disable fingerprint lock & then passcode lock.
3. Exit settings & Close Mega app from Android recent task list view.
4. Open Mega app.
5. Enable passcode lock (alphanumerical)
6. Exit settings & Close Mega app from Android recent task list view.
7. Open Mega app.
= No prompt for passcode.
8. Enable fingerprint lock.
9. Exit settings & Close Mega app from Android recent task list view.
10. Open Mega app.
= No prompt for fingerprint, nor passcode.

After this I tried the same steps but with 4 digit PIN instead of alphanumerical, but it still didn't work. I also tried with "immediately" and "30 seconds" for the "require passcode" setting, no change, still doesn't work.

I'm using Android 9.

**Comment by repsorp** [comment]
> Updated to version 10.0.3 and still doesn't work! Not passcode, nor fingerprint lock.
> 
> Steps I did to test this after update:
> 
> 1. Open Mega app.
> 2. Disable fingerprint lock & then passcode lock.
> 3. Exit settings & Close Mega app from Android recent task list view.
> 4. Open Mega app.
> 5. Enable passcode lock (alphanumerical)
> 6. Exit settings & Close Mega app from Android recent task list view.
> 7. Open Mega app.
> = No prompt for passcode.
> 8. Enable fingerprint lock.
> 9. Exit settings & Close Mega app from Android recent task list view.
> 10. Open Mega app.
> = No prompt for fingerprint, nor passcode.
> 
> After this I tried the same steps but with 4 digit PIN instead of alphanumerical, but it still didn't work. I also tried with "immediately" and "30 seconds" for the "require passcode" setting, no change, still doesn't work.
> 
> I'm using Android 9.

Have you tried uninstalling/reinstalling the Mega application?
This is personally what I did.

**Comment by Perkolator** [comment]
> Have you tried uninstalling/reinstalling the Mega application?
> This is personally what I did.

You did that after updating to version 10.0.3? It started to work for you only after reinstall? Did you try the same with previous updates?

**Comment by repsorp** [comment]
I did it the first time after noticing the issue...

**Comment by Perkolator** [comment]
Reinstalling didn't work. This is what I tried:

1. Log out & log back in.
2. Set passcode lock.
3. Exit settings & Close Mega app from Android recent task list view.
4. Open Mega app.
= No prompt for passcode.
5. Log out.
6. Cleared cache & storage for Mega app.
7. Uninstalled & reinstalled Mega app.
8. Open Mega app.
9. After log in, allowed storage & camera permissions in the setup process.
10. Set passcode lock (alphanumerical). No other settings touched.
11. Exit settings & Close Mega app from Android recent task list view.
12. Open Mega app.
= No prompt for passcode.
13. Enable fingerprint lock.
14. Exit settings & Close Mega app from Android recent task list view.
15. Open Mega app.
= No prompt for fingerprint, nor passcode.

It just doesn't work anymore. Maybe Mega devs silently dropped support for Android 9?

**Comment by repsorp** [comment]
> It just doesn't work anymore. Maybe Mega devs silently dropped support for Android 9?

hard to know precisely, not much info in the changelog and they are not very responsive here...

**Comment by Perkolator** [comment]
> not much info in the changelog and they are not very responsive here...

No kidding.. it's a shame that this repo is so badly run, and for so long. I've been thinking of canceling my yearly Pro II subscription just because of this. I also fear about the upcoming big sync client update, it might be a big mess.. I don't find the messages in the repo issue that follows its development that encouraging (it's also pretty much the only issue where actually somebody from mega is actively answering people).

> Did you try to report issue directly in the app ? (Settings > Report Issue)

I just did. Thanks.

**Comment by Perkolator** [comment]
@repsorp Forgot to ask, should I open a new issue? Do you want to close this issue?

**Comment by Perkolator** [comment]
> Did you try to report issue directly in the app ? (Settings > Report Issue)

Well, that didn't do any good. They just sent me an automatic reply with "try logout/login, un-install/re-install, restart device" advice, ALL steps that I mentioned in my initial report that I had already tried. Useless. Mega support is just absolutely useless. I think I'm going to switch to another cloud provider.

**Comment by repsorp** [comment]
> @repsorp Forgot to ask, should I open a new issue? Do you want to close this issue?

Since this issue has not been solved in your case I guess we could still let it open.

But if you prefer to open a new one a bit more specific you're welcome to do as you feel.

**Comment by repsorp** [comment]
Considering the fact that the problem was resolved in my case I think that it is actually better to close this issue.
